services/autoscaling/groups.go

```diff
@@ -0,0 +1,544 @@
+	} else {
+		if value, ok := d.GetOk("datacenter_id"); ok {
+			value := value.(string)
+			datacenter := autoscaling.GroupPutPropertiesDatacenter{
+				Id: &value,
+			}
+			group.Properties.Datacenter = &datacenter
+		}
+	}
+	return &group, nil
+}
+
+func GetAutoscalingGroupPolicyData(d *schema.ResourceData) *autoscaling.GroupPolicy {
+
+	groupPolicy := autoscaling.GroupPolicy{}
+
+	if value, ok := d.GetOk("policy.0.metric"); ok {
+		value := autoscaling.Metric(value.(string))
+		groupPolicy.Metric = &value
+	}
+
+	if value, ok := d.GetOk("policy.0.range"); ok {
+		value := value.(string)
+		groupPolicy.Range = &value
+	} else {
+		groupPolicy.Range = nil
+	}
+
+	groupPolicy.ScaleInAction = GetScaleInActionData(d)
+
+	if value, ok := d.GetOk("policy.0.scale_in_threshold"); ok {
+		value := float32(value.(int))
+		groupPolicy.ScaleInThreshold = &value
+	}
+
+	groupPolicy.ScaleOutAction = GetScaleOutActionData(d)
+
+	if value, ok := d.GetOk("policy.0.scale_out_threshold"); ok {
+		value := float32(value.(int))
+		groupPolicy.ScaleOutThreshold = &value
+	}
+
+	if value, ok := d.GetOk("policy.0.unit"); ok {
+		value := autoscaling.QueryUnit(value.(string))
+		groupPolicy.Unit = &value
+	}
+
+	return &groupPolicy
+}
+
+func GetScaleInActionData(d *schema.ResourceData) *autoscaling.GroupPolicyScaleInAction {
+	scaleInAction := autoscaling.GroupPolicyScaleInAction{}
+
+	if value, ok := d.GetOk("policy.0.scale_in_action.0.amount"); ok {
+		value := float32(value.(int))
+		scaleInAction.Amount = &value
+	}
+
+	if value, ok := d.GetOk("policy.0.scale_in_action.0.amount_type"); ok {
+		value := autoscaling.ActionAmount(value.(string))
+		scaleInAction.AmountType = &value
+	}
+
+	if value, ok := d.GetOk("policy.0.scale_in_action.0.termination_policy_type"); ok {
+		value := autoscaling.TerminationPolicyType(value.(string))
+		scaleInAction.TerminationPolicy = &value
+	}
+
+	if value, ok := d.GetOk("policy.0.scale_in_action.0.cooldown_period"); ok {
+		value := value.(string)
+		scaleInAction.CooldownPeriod = &value
+	} else {
+		scaleInAction.CooldownPeriod = nil
+	}
+
+	if value, ok := d.GetOk("policy.0.scale_in_action.0.delete_volumes"); ok {
+		value := value.(bool)
+		scaleInAction.DeleteVolumes = &value
+	} else {
+		scaleInAction.DeleteVolumes = nil
+	}
+
+	return &scaleInAction
+}
+
+func GetScaleOutActionData(d *schema.ResourceData) *autoscaling.GroupPolicyScaleOutAction {
+	scaleOutAction := autoscaling.GroupPolicyScaleOutAction{}
+
+	if value, ok := d.GetOk("policy.0.scale_out_action.0.amount"); ok {
+		value := float32(value.(int))
+		scaleOutAction.Amount = &value
+	}
+
+	if value, ok := d.GetOk("policy.0.scale_out_action.0.amount_type"); ok {
+		value := autoscaling.ActionAmount(value.(string))
+		scaleOutAction.AmountType = &value
+	}
+
+	if value, ok := d.GetOk("policy.0.scale_out_action.0.cooldown_period"); ok {
+		value := value.(string)
+		scaleOutAction.CooldownPeriod = &value
+	} else {
+		scaleOutAction.CooldownPeriod = nil
```

Comment:
ScaleOutAction cooldown_period has a default value, so this else won't ever be triggered.

Reply:
@cbeti-ionos solved in c9df22e0066ca69a7809de59cd124fe97fec9580